compiler/testData/debug/stepping/namedCallableReference.kt

```diff
@@ -10,10 +10,19 @@ fun f(block: () -> Unit) {
 
 fun g() {}
 
+// The synthetic invoke bridge method generated for in the callable reference has line numbers
+// in the JVM_IR backend (as all bridges). In the JVM backend, only some bridges have line numbers.
+// For some reason, when the bridge does not have line numbers, there is no method entry event
+// for the invoke method bridged to. Therefore, the entry line number for invoke only shows
+// up for JVM_IR.
+
 // LINENUMBERS
 // test.kt:3 box
 // test.kt:4 box
 // test.kt:8 f
+// LINENUMBERS JVM_IR
+// test.kt:4 invoke
```

Comment:
@madsager Is this test affected by #3735 or not?

Reply:
It is not, no.

The difference in this test comes from the `invoke` bridge that returns an `Object` having line numbers with JVM_IR and not with JVM.

JVM:

```
  public java.lang.Object invoke();
    descriptor: ()Ljava/lang/Object;
    flags: ACC_PUBLIC, ACC_BRIDGE, ACC_SYNTHETIC
    Code:
      stack=1, locals=1, args_size=1
         0: aload_0
         1: invokevirtual #12                 // Method invoke:()V
         4: getstatic     #18                 // Field kotlin/Unit.INSTANCE:Lkotlin/Unit;
         7: areturn

  public final void invoke();
    descriptor: ()V
    flags: ACC_PUBLIC, ACC_FINAL
    Code:
      stack=0, locals=1, args_size=1
         0: invokestatic  #23                 // Method FlafKt.g:()V
         3: return
      LineNumberTable:
        line 4: 0
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
            0       4     0  this   LFlafKt$box$1;
```

JVM_IR
```
  public final void invoke();
    descriptor: ()V
    flags: ACC_PUBLIC, ACC_FINAL
    Code:
      stack=0, locals=1, args_size=1
         0: invokestatic  #23                 // Method FlafKt.g:()V
         3: return
      LineNumberTable:
        line 4: 0
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
            0       4     0  this   LFlafKt$box$1;

  public java.lang.Object invoke();
    descriptor: ()Ljava/lang/Object;
    flags: ACC_PUBLIC, ACC_BRIDGE, ACC_SYNTHETIC
    Code:
      stack=1, locals=1, args_size=1
         0: aload_0
         1: invokevirtual #27                 // Method invoke:()V
         4: getstatic     #32                 // Field kotlin/Unit.INSTANCE:Lkotlin/Unit;
         7: areturn
      LineNumberTable:
        line 4: 0
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
            0       8     0  this   LFlafKt$box$1;
```

I must admit that I'm confused as to why the debugging filters do not stop on line 4 in the usual non-bridge `invoke` method for JVM. There must be some kind of filtering going on.

Reply:
@yanex @goodwinnk @NataliaUkhorskaya Maybe you now what is going on with filtering?